### PR TITLE
feat: REPLAT-9476 metadata fix

### PR DESCRIPTION
**Pull request by Daniel-Georgiev**
This PR is for https://nidigitalsolutions.jira.com/secure/RapidBoard.jspa?rapidView=1117&projectKey=REPLAT&modal=detail&selectedIssue=REPLAT-9476.
Added additional conditions if any of the props (bylines, content, shortHeadline, template) are null.
Added snapshots and tests.

### CODE REVIEW

File: packages/article-skeleton/__tests__/web/head.web.test.js [hunk 1]
```diff
@@ -87,6 +87,22 @@ describe("Head", () => {
     expect(title.children).toMatchSnapshot();
   });
 
+  it("defaults to an empty string if both headline and shortHeaddline are null", () => {
```
Comment (aronshamash) [inline packages/article-skeleton/__tests__/web/head.web.test.js]: typo

File: packages/article-skeleton/src/article-skeleton.web.js [hunk 2]
```diff
@@ -79,8 +72,8 @@ class ArticleSkeleton extends Component {
     } = article;
 
     const { articleWidth } = this.state;
-    const newContent = maybeInsertDropcap(
```
Comment (aronshamash) [inline packages/article-skeleton/src/article-skeleton.web.js]: nice!

File: packages/article/src/article.js [hunk 3]
```diff
@@ -26,7 +26,12 @@ export class TakeoverBailout extends Error {
 
 const Article = props => {
   const { article, onImagePress } = props;
-  const { leadAsset, template = "mainstandard" } = article || {};
+  if(article){
```
Comment (aronshamash) [inline packages/article/src/article.js]: think this will fail linting

File: packages/responsive-image/package.json [hunk 4]
```diff
@@ -26,7 +26,7 @@
     "test": "jest",
     "test:lint": "tslint --project . && prettier \"src/**/*.ts\" --list-different",
     "test:unit": "jest --coverage",
-    "watch": "run-s clean build:main && run-p \"build:main -- -w\" \"test:unit -- --watch\"",
+    "watch": "run-s clean build:module && run-p \"build:module -- -w\" \"test:unit -- --watch\"",
```
Comment (aronshamash) [inline packages/responsive-image/package.json]: think this is already merged ?

Comment (Daniel-Georgiev) [reply packages/responsive-image/package.json]: yep!

File: packages/article/src/article.js [hunk 5]
```diff
@@ -26,7 +26,11 @@ export class TakeoverBailout extends Error {
 
 const Article = props => {
   const { article, onImagePress } = props;
-  const { leadAsset, template = "mainstandard" } = article || {};
+  if (article) {
+    article.template = article.template || "mainstandard";
+  }
+
```
Comment (apostolnikov) [inline packages/article/src/article.js]: Maybe we should use some object assignment here as mutating the props directly on a component level is never a good idea and can lead to a multiple bugs and problems.

Comment (Daniel-Georgiev) [reply packages/article/src/article.js]: Looking into it!

File: packages/styleguide/src/theme/theme-factory.js [hunk 6]
```diff
@@ -52,17 +45,22 @@ const magazineFontPicker = (
   return config[template][section] || defaultFont;
 };
 
-const headlineCasePicker = (section = "default", template = "mainstandard") =>
```
Comment (aronshamash) [inline packages/styleguide/src/theme/theme-factory.js]: basically we need to remove nearly all default params from the codebase as it is doesn't work when values are `null`. So back to the old ways ...

File: packages/styleguide/src/theme/theme-factory.js [hunk 7]
```diff
@@ -52,17 +45,22 @@ const magazineFontPicker = (
...
-  dropCapFont: magazineFontPicker("dropCap", section, template),
-  headlineFont: magazineFontPicker("headline", section, template),
-  pullQuoteFont: magazineFontPicker("headlineRegular", section, template),
-  sectionColour: sectionColourPicker(section, template),
-  headlineCase: headlineCasePicker(section, template)
-});
+export default (sectionParam, templateParam) => {
+  const section = sectionParam || "default";
+  const template = templateParam || "mainstandard";
+
```
[the remark this answers is not in the document]
Comment (aronshamash) [reply packages/styleguide/src/theme/theme-factory.js]: much nicer

**Pull request merged**